ddtrace/appsec/context_vars.py

```diff
@@ -0,0 +1,17 @@
+from ddtrace.vendor import contextvars
+
+
+# FIXME: remove these and use the new context API once implemented and allowing
+# contexts without spans
+
+_DD_EARLY_IP_CONTEXTVAR = contextvars.ContextVar("datadog_early_ip_contextvar", default=None)
+_DD_EARLY_HEADERS_CONTEXTVAR = contextvars.ContextVar("datadog_early_headers_contextvar", default=None)
+_DD_EARLY_HEADERS_CASE_SENSITIVE_CONTEXTVAR = contextvars.ContextVar(
+    "datadog_early_headers_casesensitive_contextvar", default=False
+)
```

Comment:
I kept the functions separate in case we need them in the future (I could see some blocking based on user agent or other header values as a future feature) but took your first idea and created the context manager with all the parameters (then it individually calls each function inside).